php_pthreads.c

```diff
@@ -156,11 +156,11 @@ static inline void pthreads_execute_ex(zend_execute_data *data) {
 /* {{{ */
 static inline zend_bool pthreads_verify_type(zend_execute_data *execute_data, zval *var, zend_arg_info *info) {
 	pthreads_object_t *threaded = NULL;
-
+        
 	if (!var ||
 #if PHP_VERSION_ID >= 70200
-		!ZEND_SAME_FAKE_TYPE(ZEND_TYPE_CODE(info->type), Z_TYPE_P(var)) &&
-		!ZEND_TYPE_IS_CLASS(info->type) ||
+                Z_ISNULL_P(var) ||
```

Comment:
A change from `&&` to `||` at that point would work for primitive types(fake type?) only, but not for classes. In case of an object/class type, `ZEND_SAME_FAKE_TYPE` would let the condition match and falsely return. In generally the `ZEND_SAME_FAKE_TYPE` check is unnecessary in `pthreads_verify_type`, because `pthreads_verify_type` is only called when the types doesn't match. If the type is a class, do `instanceof_function`. In every other case, type is a primitive or a null var, return immediately and let the script fail.